src/services/codefixes/importFixes.ts

```diff
@@ -1364,13 +1386,11 @@ function doAddExistingFix(
 
     const promoteFromTypeOnly = clause.isTypeOnly && some([defaultImport, ...namedImports], i => i?.addAsTypeOnly === AddAsTypeOnly.NotAllowed);
     const existingSpecifiers = clause.namedBindings && tryCast(clause.namedBindings, isNamedImports)?.elements;
-    // If we are promoting from a type-only import and `--isolatedModules` and `--preserveValueImports`
-    // are enabled, we need to make every existing import specifier type-only. It may be possible that
-    // some of them don't strictly need to be marked type-only (if they have a value meaning and are
-    // never used in an emitting position). These are allowed to be imported without being type-only,
-    // but the user has clearly already signified that they don't need them to be present at runtime
-    // by placing them in a type-only import. So, just mark each specifier as type-only.
-    const convertExistingToTypeOnly = promoteFromTypeOnly && importNameElisionDisabled(compilerOptions);
+    // We used to convert existing specifiers to type-only only if compiler options indicated that
+    // would be meaningful (see the `importNameElisionDisabled` utility function), but user
+    // feedback indicated a preference for preserving the type-onlyness of existing specifiers
+    // regardless of whether it would make a difference in emit.
+    const convertExistingSpecifiersToTypeOnly = promoteFromTypeOnly;

```

Comment:
Does this constant need to stay? It's already changing names, so it seems like the uses could just be updated to be `promoteFromTypeOnly`, unless there's something I'm missing that's not shown in the diff.

Reply:
It doesn’t, but looking at where it was used, I thought it was more clear to keep it.